msg_queue.go

```diff
@@ -8,17 +8,78 @@ import (
 // msgQueue defines the structure that holds message queues for different PBFT states
 type msgQueue struct {
 	// Heap implementation for the round change message queue
-	roundChangeStateQueue msgQueueImpl
+	roundChangeStateQueue *msgQueueImpl
 
 	// Heap implementation for the accept state message queue
-	acceptStateQueue msgQueueImpl
+	acceptStateQueue *msgQueueImpl
 
 	// Heap implementation for the validate state message queue
-	validateStateQueue msgQueueImpl
+	validateStateQueue *msgQueueImpl
+
+	pendingCommitStateQueue *msgQueueImpl
+
+	commitValidation *commitValidationRoutine
 
 	queueLock sync.Mutex
 }
 
+type commitValidationRoutine struct {
+	updateCh chan *RoundInfo2
+	closeCh  chan struct{}
+	queue    *msgQueueImpl
+}
+
+func (c *commitValidationRoutine) init() {
+	var round *RoundInfo2
+
```

Comment:
workCh := make(chan Work)
worker := func() {
   for _, work := <-workCh {
     ...
   }
}

for i := 0; i < 5; i++ {
  go worker()
}